Jenkinsfile

```diff
@@ -142,114 +205,192 @@ builders['msvc2015'] = {node('windows'){
 
 builders['ubuntu-16.04'] = { node ('ubuntu-16.04'){
     timestamps {
-      try {
-        stage('ubuntu 16.04') {
+      stage('checkout ubuntu 16.04') {
+        try {
           checkout scm
           sh 'rm -Rf build'
-          dir ("build") {
-            def branch = getBranch()
-            sh """cmake -DGOOGLETEST_SRC_PATH=\"${tool 'googletest'}\" -DCMAKE_BUILD_TYPE=${buildConfiguration} .."""
-            sh "cmake --build . --config ${buildConfiguration} -- -j3 > error_and_warnings.txt 2>&1"
-            sh "cat error_and_warnings.txt"
-            sh "cpack -D CPACK_PACKAGE_FILE_NAME=veles-ubuntu1604 -G \"ZIP;DEB\" -C ${buildConfiguration}"
-            junit allowEmptyResults: true, keepLongStdio: true, testResults: '**/results.xml'
-            step([$class: 'ArtifactArchiver', artifacts: 'veles-ubuntu1604.*', fingerprint: true])
-            step([$class: 'WarningsPublisher', canComputeNew: false, canResolveRelativePaths: false, defaultEncoding: '',
-            excludePattern: '', healthy: '', includePattern: '', messagesPattern: '',
-            parserConfigurations: [[parserName: 'GNU Make + GNU C Compiler (gcc)', pattern: 'error_and_warnings.txt']], unHealthy: ''])
-          }
-          sh 'rm -Rf build'
+        } catch (error) {
+          post_stage_failure(env.JOB_NAME, "Ubuntu checkout", env.BUILD_ID, env.BUILD_URL)
+          throw error
         }
-      } catch (error) {
-        sh "cat build/error_and_warnings.txt"
-        post_stage_failure(env.JOB_NAME, "ubuntu-16.04-amd64", env.BUILD_ID, env.BUILD_URL)
-        throw error
       }
+      parallel(
+        "build ubuntu 16.04": {
+          stage('ubuntu 16.04') {
+            try {
+              dir ("build") {
+                def branch = getBranch()
+                sh """cmake -DGOOGLETEST_SRC_PATH=\"${tool 'googletest'}\" -DCMAKE_BUILD_TYPE=${buildConfiguration} .."""
+                sh """#!/bin/bash -ex
+                  set -o pipefail
+                  cmake --build . --config ${buildConfiguration} -- -j3 2>&1 | tee error_and_warnings.txt
+                """
+                sh """#!/bin/bash -ex
+                  set -o pipefail
+                  cmake --build . --target lint --config ${buildConfiguration} -- -j3 2>&1 | tee lint_error_and_warnings.txt
```

Comment:
`lint_error_and_warnings` -> `lint_errors_and_warnings` (please fix this also in other places)